tools/database/dbc/spell_effect.go

```diff
@@ -207,15 +207,34 @@ func (data *SpellEffect) ClassFlag(index uint) uint32 {
 	return uint32(data.EffectSpellClassMasks[index/32]) & (1 << (index % 32))
 }
 
-func (effect *SpellEffect) ParseStatEffect() *stats.Stats {
+func (effect *SpellEffect) ParseStatEffect(scalesWithIlvl bool, ilvl int) *stats.Stats {
 	stats := &stats.Stats{}
 	scale := effect.ScalingClass()
-	spell := dbcInstance.Spells[effect.ID]
+	spell := dbcInstance.Spells[effect.SpellID]
 	stat, _ := MapMainStatToStat(effect.EffectMiscValues[0])
 
 	switch {
+	case effect.EffectAura == A_MOD_RANGED_ATTACK_POWER:
+		if scalesWithIlvl && effect.Coefficient != 0 && spell.Attributes[8]&0x1000 != 0 {
+			propPoints := dbcInstance.RandomPropertiesByIlvl[ilvl][proto.ItemQuality_ItemQualityEpic][0] // Epic 0 for items
+			stats[proto.Stat_StatRangedAttackPower] = math.Floor(float64(propPoints) * effect.Coefficient)
```

Comment:
Likewise, define a helper for this calculation so you're only coding the logic once:
```
func (effect *SpellEffect) CalcRandomPropStatValue(ilvl int) float64 {
    propPoints := dbcInstance.RandomPropertiesByIlvl[ilvl][proto.ItemQuality_ItemQualityEpic][0] // Epic 0 for items
    return math.Floor(float64(propPoints) * effect.Coefficient)
}
```